Update to support v8.3

Bobsson opened the pull request:
See https://github.com/notepad-plus-plus/notepad-plus-plus/issues/11198 for more information.

In order to get this to build on my machine under VS 2019 I also had to add an `#include <string>` to Config.h, but I don't know enough to know whether that's actually necessary. The auto-build didn't need it.

Bobsson committed:
Update to support v8.3

See https://github.com/notepad-plus-plus/notepad-plus-plus/issues/11198 for more information

tolot27 commented:
@Bobsson I picked up the artifact and installed it but the plugin is still not listed in the plugins menu and does not work in v8.3.3. What else is needed?

rdipardo commented:
> @Bobsson I picked up the artifact and installed it but the plugin is still not listed in the plugins menu and does not work in v8.3.3. What else is needed?

8.3.3 will ignore a plugin if its version is on [the "blacklist"][0]. You have to increase the version number and rebuild:

```diff
diff --git a/src/Version.h b/src/Version.h
index 9bada61..4fda386 100644
--- a/src/Version.h
+++ b/src/Version.h
@@ -16,8 +16,8 @@
 // along with this program; if not, write to the Free Software
 // Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 
-#define VERSION_NUM				1,3,1,0
-#define VERSION_LINEAR			1310
-#define VERSION_LINEAR_TEXT		TEXT("1310")
-#define VERSION_TEXT			TEXT("1.3.1") // This must match the tag pushed on the server minus the "v"
+#define VERSION_NUM				1,3,2,0
+#define VERSION_LINEAR			1320
+#define VERSION_LINEAR_TEXT		TEXT("1320")
+#define VERSION_TEXT			TEXT("1.3.2") // This must match the tag pushed on the server minus the "v"
 #define VERSION_STAGE			TEXT("") // "alpha", "beta", ""

```

That will trick N++ into thinking it's safe:

![elastic_v132_x64_gif](https://user-images.githubusercontent.com/59004801/161931396-181d4856-8bb8-45e7-9d99-7a47fe6d5dbf.gif)

More simply, you can download a portable 64-bit version of N++ [8.3.1][1], which will load any plugin version without checking first.

[0]: https://github.com/notepad-plus-plus/nppPluginList/commit/2137a2c0cd908ca0c615712dd1d6165645cbde29
[1]: https://github.com/notepad-plus-plus/notepad-plus-plus/releases/tag/v8.3.1

tolot27 commented:
@Bobsson Please can you increase the version to at least 1.3.1.1 and push it again? Afterwards, I can retest the CI artifact. Currently, I don't have VS running.

Bobsson commented:
@rdipardo Interesting.  On my machine, N++ 8.3.3 seems to have loaded my build of 1.3.1 without a problem.  Maybe it builds differently locally?
I really should have bumped the version number anyway, though, so I'll do that and see if it helps.

Bobsson committed:
Update version number to 1.3.1.1

Bobsson committed:
Keep three-part version number (v1.3.2)

Same code as 1.3.1.1, but since the build server adds the fourth digit, this is better.

Bobsson committed:
... and another place the version number lives

dail8859 commented:
@Bobsson Thanks for the PR!

@tolot27 @rdipardo Thanks for the testing and feedback!

marko090591 commented:
So when  version number 1.3.1.1  (work with notepad++ 8.3) will be released?

dail8859 commented:
> So when  version number 1.3.1.1  (work with notepad++ 8.3) will be released? 

Whenever I get some free time to create a release and add it to the plugin list. Then have to wait for a new Notepad++ release.

tolot27 commented:
@marko090591 In the mean time you can download the [x64 artifact](https://ci.appveyor.com/project/dail8859/elastictabstops/builds/43155677/job/8aohta1g2deuc24t/artifacts).

marko090591 commented:
@Dosipago thank you for given information.

ExE-Boss commented:
@dail8859 Note that the artifacts will soon be removed from **Appveyor**[^1].

[^1]: [in 14 days](https://www.timeanddate.com/countdown/generic?iso=2022-05-06T23:00:00Z&p0=1440&msg=Countdown+Timer&font=sanserif&csz=1 "2022-05-06T23:00:00Z") at the time of writing.

Solarbotics commented:
> @marko090591 In the mean time you can download the [x64 artifact](https://ci.appveyor.com/project/dail8859/elastictabstops/builds/43155677/job/8aohta1g2deuc24t/artifacts).

Wonderful - you're a champ. Thanks for that

marko090591 commented:
@dail8859 could you re-publish the ElasticTabstops 1.3.2.11. dll at the site bellow. Thank you. 
#
https://ci.appveyor.com/project/dail8859/elastictabstops/builds/43155677/job/8aohta1g2deuc24t/artifacts

marko090591 commented:
Or mybe @tolot27. Read one post abowe and respond kindly.

tolot27 commented:
> @dail8859 could you re-publish the ElasticTabstops 1.3.2.11. dll at the site bellow. Thank you.

No, that is not possible because it is an artifact automatically build by Appveyor and will be removed after 14 days as written at https://github.com/dail8859/ElasticTabstops/pull/30#issuecomment-1106728712.

marko090591 commented:
@Bobsson could you provide a link?

dail8859 commented:
I was getting ready to publish a new release but the plugin seems to have a bug that it isn't updating things correctly.

marko090591 commented:
Yes. When data is pasted in Notepad++ you must change the tab window, to ElasticTabstops to alling the copyed data (present at older versions) . Also if a line is present, the data after blank line is not alligned correcty in reference of data before the blank line. Mentioned is little annoying.

marko090591 commented:
> I was getting ready to publish a new release but the plugin seems to have a bug that it isn't updating things correctly.

Yes. When data is pasted in Notepad++ you must change the tab window, to ElasticTabstops to alling the copyed data (present at older versions) . Also if a line is present, the data after blank line is not alligned correcty in reference of data before the blank line. Mentioned is little annoying.

dail8859 commented:
@marko090591

Not even just pasting but basic editing functionality seems broken.

dail8859 commented:
Looks like the new Scintilla version is the root cause of the random behavior I saw before publishing a new release.

https://github.com/notepad-plus-plus/notepad-plus-plus/issues/11766
